public/home.js

```diff
@@ -1,6 +1,216 @@
+          startTime: startTime,
+          date: date,
+          duration: duration,
+          meetingLocation: meetingLocation,
+          comments: comments,
+        }
+      },
+      dataType: 'json',
+      success: function(response){
+        alert("Thank you for the post!")
+      },
+      error: function(xhr, status, data){
+        console.log(xhr);
+      }
+    })
+  }
+
+  var picChoice = function(type){
+    if(type == 'SWIM'){
+      return '/public/assets/swim.jpeg'
+    } else if (type == 'BIKE'){
+      return '/public/assets/bike.jpg'
+    } else if (type == 'RUN'){
+      return '/public/assets/run.jpeg'
+    } else if (type == 'TEAM'){
+      return '/public/assets/team.jpeg'
+    } else if (type == 'GROUP'){
+      return '/public/assets/bootcamp.jpeg'
+    } else if (type == 'OTHER') {
+      return '/public/assets/others.jpg'
+    }
+    
+  }
+
+
+  APIaction.prototype.getPostSuccess = function(response){
+    var constructHTML = function(response){
+      var html = '';
+
+      for (i = 0; i < response.length; i++){
+      html +=  '<div class="col-md-4">'
+      html +=    '<div class="thumbnail img-responsive">'
+      html +=      '<img src=' + picChoice(response[i].message.type)+'>'
+      html +=        '<div class="caption">'
+      html +=          '<h3>Title: '+response[i].message.title+'</h3>'
+      html +=          '<p>Type: '+response[i].message.type+'</p>'
+      html +=          '<p>Date:' +new Date(response[i].message.date).toISOString().slice(0, 10)+'</p>'
+      html +=          '<p>Time:' +response[i].message.startTime+'</p>'
+      html +=          '<p>Duration:' +response[i].message.duration+'</p>'
+      html +=          '<p>Meet Pt:' +response[i].message.meetingLocation+'</p>'
+      html +=          '<p><button class="btn btn-success" role="button">Join</button>'
+      html +=          '<button class="btn btn-danger" role="button">Unjoin</button>'
+      html +=          '<button type="button" class="btn btn-primary more-info" data-container="body" data-toggle="popover" data-placement="bottom" data-content='+response[i].message.comments+'>MORE</button></p>'   
+      html +=         '</div>'
+      html +=     '</div>'
+      html +=  '</div>'
+      }
+      return html;
+     
+    };
+
+    html = constructHTML(response);
+    $('.workOutPosts').html(html);
+
+  }
+
+  APIaction.prototype.getAllPost = function(){
+    $.ajax({
+      type: 'GET',
+      url: '/workouts',
+      dataType: 'json',
+      success: this.getPostSuccess
+      
+
+    })
+  }
+
+  APIaction.prototype.pageRefresh = function(){
+    return setInterval(this.getAllPost(),5000)
+  }
+
+  var apiAction = new APIaction ();
+  
+  //name display
+  apiAction.nameDisplay();
+  
+  //autoRefresh
+  apiAction.pageRefresh();
+
+  //refresh all post (manually)
+    $('#refresh').on('click', function(){
+    apiAction.getAllPost()
+  })
+
+  //create workout post
+  $('#confirmPost').on('click',function(){
```

Comment:
you could consider also using [submit](https://api.jquery.com/submit/) and adding attributes to the [input](https://developer.mozilla.org/en/docs/Web/HTML/Element/Input#Attributes) such as `required` or `maxlength`... just an idea